### DISCUSSION TITLE
Rendering of SDFs

### DISCUSSION THREAD

**Discussion by theottm** [kaveh808/kons-9#252] [Ideas]
Raymarching a SDF directly is faster that sampling it to a mesh and rendering that. I even heard that raymarching an SDF is faster than raytracing triangles from an existinh mesh for photorealistic rendering (lots of bounces). Is that true?

The only raymarching algorithm I know is basic sphere marching. Maybe there are better approaches?
One thing I always wondered was if one could use the gradient of the SDF (so a vector field) to optimize the number of steps (for example when a ray goes almost parallel to a surface).

One big question is how to pass our SDF to a fragment shader, so how to convert CL to GLSL. These might help: [varjo](https://github.com/cbaggers/varjo), [3bgl-shader](https://github.com/3b/3bgl-shader).

**Comment by ghost** [comment]


**Reply by theottm** [reply]
Nice, thanks for the references, I'll try to dig into that later. I already implemented a simple sphere tracer that just works in shadertoy, but I'm curious about going at it with a precise theoretical framework.

**Reply by theottm** [reply]
I don't know the SDF of an immersed Klein bottle, but I tried rendering quadrics, using their defining quadratic equation as a "SDF", turned out it doesn't work well :)

**Comment by ghost** [comment]


**Reply by theottm** [reply]
Yes it's not the biggest of our questions probably :)  we can do it

**Comment by ghost** [comment]


**Reply by theottm** [reply]
Thanks for sharing! I'll have a deeper look when I start playing with an implementation

**Comment by ghost** [comment]


**Reply by theottm** [reply]
I get it. Keep going ;)
